### PR TITLE
fix: mlsthumbprint derivation [WPB-6878]

### PR DESCRIPTION
**Pull request by markusfassbender**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-6878" title="WPB-6878" target="_blank">BugWPB-6878</a>  [iOS] Mismatch MLS Thumbprint 
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Derive the mls thumbprint from the certificate instead of having multiple parameters, that might cause unwanted differences between data and are harder to follow.

The logic itself didn't change, only the way objects are passed.

### Solutions

Remove convenience and pass the values, that are needed.

#### Test Coverage (Optional)

- [x] I have added automated test to this contribution

#### How to Test

- Run unit test suite

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: wire-ios/Wire-iOS/Sources/UserInterface/Settings/DeviceManagement/ClientListViewController.swift [hunk 1]
```diff
@@ -228,15 +219,47 @@ final class ClientListViewController: UIViewController,
...
             self.navigationController?.setNavigationBarHidden(false, animated: false)
         }
         let hostingViewController = UIHostingController(rootView: detailsView)
         hostingViewController.view.backgroundColor = SemanticColors.View.backgroundDefault
+
         navigationController.pushViewController(hostingViewController, animated: true)
         navigationController.isNavigationBarHidden = true
     }
 
+    private func makeDeviceInfoViewModel(
```
Comment (theswiftway) [inline wire-ios/Wire-iOS/Sources/UserInterface/Settings/DeviceManagement/ClientListViewController.swift:233]: **suggestion:** Do you think making this into an init method makes more sense?

File: wire-ios/Wire-iOS/Sources/UserInterface/UserProfile/Devices/UserClientListViewController.swift [hunk 2]
```diff
@@ -192,6 +181,38 @@ final class UserClientListViewController: UIViewController,
         navigationController.pushViewController(hostingViewController, animated: true)
         navigationController.isNavigationBarHidden = true
     }
+
+    private func makeDeviceInfoViewModel(
```
Comment (theswiftway) [inline wire-ios/Wire-iOS/Sources/UserInterface/UserProfile/Devices/UserClientListViewController.swift:185]: **suggestion:** Do you think making this into an init method makes more sense?

File: wire-ios/Wire-iOS/Sources/UserInterface/Settings/DeviceManagement/ClientListViewController.swift (hunk 1 above)
Comment (markusfassbender) [reply wire-ios/Wire-iOS/Sources/UserInterface/Settings/DeviceManagement/ClientListViewController.swift:233]: Thanks, but I think it's okay the way it is. If one starts to make it more generic / abstract another level it looks simple in the beginning, but small details like `isFromConversation: false` needs to be configurable and in the end there no real benefit.

This case, I just removed, is the best example:
Screenshot 2024-03-13 at 14 10 15

**Review by theswiftway — APPROVED** [review]
LGTM

**Review by caldrian — APPROVED** [review]

**Pull request merged**